src/model.py

```diff
@@ -2048,22 +2048,21 @@ def init_components(self, subtasks):
         relations = torch.nn.ModuleDict({})
         attn_linears = torch.nn.ModuleDict({})
         for stack in range(stack_count):
+            relations[f"stack_{stack}"] = torch.nn.ModuleDict({})
             for rel in attention_relations:
-                relations[f"stack_{stack}"] = torch.nn.ModuleDict({
-                    f"{rel[0]}_at_{rel[1]}": torch.nn.ModuleDict({
-                        "attn": self.attn_block(cnn_dim),
+                relations[f"stack_{stack}"][f"{rel[0]}_at_{rel[1]}"] = torch.nn.ModuleDict({
+                    "attn": self.attn_block(cnn_dim),
 
-                        # new linear for each attn relation for first_task logits using attn information
-                        "linear": self.linear_block(cnn_dim*2, 3),
-                    })
+                    # new linear for each attn relation for first_task logits using attn information
+                    "linear": self.linear_block(cnn_dim*2, 3)
```

Comment:
Notice this input size remains the same. Due to attention output w/ cnn output (no embeddings here)